consensus/src/lib.rs

```diff
@@ -34,29 +34,33 @@ pub enum ProgressResult {
 }
 
 #[async_trait]
-pub trait Consensus {
+pub trait Consensus: Send + Sync {
     async fn create(
         directory: &str,
         height: BlockHeight,
         validator_set: &[(PublicKey, VotingPower)],
     ) -> Result<(), Error>;
 
-    async fn read(directory: &str) -> Result<ConsensusState, Error>;
+    async fn new(directory: &str) -> Result<Self, Error>
+    where
+        Self: Sized;
+
+    async fn read(&self) -> Result<ConsensusState, Error>;
 
     async fn veto_block(
-        directory: &str,
+        &mut self,
         network_config: NetworkConfig,
         known_peers: &[Peer],
         block_hash: Hash256,
-        height_to_assert: BlockHeight,
     ) -> Result<(), Error>;
 
+    async fn set_proposal(&mut self, block_hash: Hash256) -> Result<(), Error>;
```

Comment:
What this function does? Marking a block to be proposed when `progress` is called?

Reply:
yes.

Reply:
thanks